.pre-commit-config.yaml

```diff
@@ -5,6 +5,13 @@ repos:
     -   id: astyle_py
         args: ['--style=otbs', '--attach-namespaces', '--attach-classes', '--indent=spaces=4', '--convert-tabs', '--align-pointer=name', '--align-reference=name', '--keep-one-line-statements', '--pad-header', '--pad-oper']
 
+-   repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+    - id: codespell
+      types_or: [c, c++]
```

Comment:
Would it make sense to add markdown files too? such as readmes?

Reply:
Yes, sure. 
I didn't expect there will be so much typos in usb host class drivers....

I have added just the source files, but I think we can add almost everything from the project.

Reply:
@tore-espressif Done. Now the codespell applies on all files without exceptions. I tried to find any files that we do not need to check, but I didn't come up with anything. 

Do you think there is something we **_don't need to verify_**?

Reply:
We don't need to verify files that are not intended for users, such as tests or repo infrastrucutre files. But in this case it might be easier to just spell check all files. LGTM!